formula/src/main/java/com/instacart/formula/FormulaRuntime.kt

```diff
@@ -166,12 +166,9 @@ class FormulaRuntime<Input : Any, Output : Any>(
             }
         }
 
-        if (effects.isNotEmpty() || evaluate) {
-            if (isRunEnabled) {
-                run(evaluate = evaluate)
-            } else {
-                pendingEvaluation = pendingEvaluation || evaluate
-            }
+        pendingEvaluation = pendingEvaluation || evaluate
```

Comment:
Added `runIfNeeded` that checks effects and pending evaluation (shared with executeBatch)